packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/search/engine/intelligence.py

```
@@ -0,0 +1,553 @@
+document relationship analysis, similarity detection, conflict detection,
+complementary content discovery, and document clustering.
+"""
+
+from typing import Any, TYPE_CHECKING
+from contextlib import contextmanager
+
+if TYPE_CHECKING:
+    from .core import SearchEngine
+
+from ...utils.logging import LoggingConfig
+from ..components.search_result_models import HybridSearchResult
```

Ruff (0.12.2)

`..components.search_result_models.HybridSearchResult` imported but unused

Remove unused import: `..components.search_result_models.HybridSearchResult`

(F401)